oauthenticator/mediawiki.py

```diff
@@ -116,6 +120,23 @@ class MWOAuthenticator(OAuthenticator):
         help='Full path to index.php of the MW instance to use to log in'
     )
 
+    pass_secrets = Bool(
+        False,
+        config=True,
+        help='Pass OAuth consumer and access secrets to the spawner'
+    )
+
+    def pre_spawn_start(self, user, spawner):
+        if not self.pass_secrets:
+            return
+
+        spawner.env.update({
+            'CLIENT_SECRET': self.client_secret,
+            'CLIENT_ID': self.client_id,
```

Comment:
It's probably not right to pass the Hub's own OAuth id and secret to all users, is it? This would mean that all users have the ability to access MW *as the Hub*.